srcjs/input_binding_date.js

```diff
@@ -212,7 +217,14 @@ $.extend(dateInputBinding, {
     if (isNaN(d))
       return null;
 
-    return new Date(d.getTime());
+    return d;
+  },
+  // A Date can have any time during a day; this will return a new Date object
+  // set to 00:00 in UTC.
+  _normalizeDate: function(date) {
```

Comment:
Maybe `_roundDate` would be better name (normalize could be referring to timezone)?